### PR TITLE
feat: Add tests for notification service extension SQSERVICES-1311 .

### PR DESCRIPTION
**Pull request by KaterinaWire**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/SQSERVICES-1311" title="SQSERVICES-1311" target="_blank">TaskSQSERVICES-1311</a>  Add tests for notification service
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Add tests for notification service extension.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: Wire-iOS Tests/Notification Service Extension/NotificationServiceTests.swift [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+//
+
+import Foundation
+import XCTest
+
+final class NotificationServiceTests: XCTestCase {
+
+    var sut: NotificationService!
+    var request: UNNotificationRequest!
+    var notificatioContent: UNNotificationContent!
```
Comment (johnxnguyen) [inline Wire-iOS Tests/Notification Service Extension/NotificationServiceTests.swift:26]: Typo: `notificatioContent` -> `notificationContent`

File: Wire-iOS Tests/Notification Service Extension/NotificationServiceTests.swift [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+        contentResult = nil
+
+        coreDataFixture = nil
+        currentUserIdentifier = nil
+        mockConversation = nil
+
+        super.tearDown()
+    }
+
+    func testThatNotificationSessionGeneratesNotification() {
```
Comment (johnxnguyen) [inline Wire-iOS Tests/Notification Service Extension/NotificationServiceTests.swift:81]: Perhaps: `testThatItHandlesGeneratedNotification`

File: Wire-iOS Tests/Notification Service Extension/NotificationServiceTests.swift [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+
+        // WHEN
+        XCTAssertFalse(callEventProviderMock.reportIncomingVoIPCallCalled)
+        sut.reportCallEvent(event, currentTimestamp: Date().timeIntervalSince1970)
+
+        // THEN
+        XCTAssertTrue(callEventProviderMock.reportIncomingVoIPCallCalled)
+    }
+
+    func testThatItDoesNotReportCallEventIfPayloadIsWrong() {
```
Comment (johnxnguyen) [inline Wire-iOS Tests/Notification Service Extension/NotificationServiceTests.swift:108]: Perhaps `testThatItDoesNotReportCallEventForNonCallEvent`

File: Wire Notification Service Extension/NotificationService.swift [hunk 4]
```diff
@@ -26,10 +26,32 @@ import WireSyncEngine
 import UIKit
 import CallKit
 
+protocol CallEventHandlerProtocol {
+    func reportIncomingVoIPCall(_ payload: [String: Any])
+}
+
+class CallEventProvider: CallEventHandlerProtocol {
```
[the remark this answers is not in the document]
Comment (KaterinaWire) [reply Wire Notification Service Extension/NotificationService.swift]: Done

File: Wire-iOS Tests/Notification Service Extension/NotificationServiceTests.swift (hunk 2 above)
Comment (KaterinaWire) [reply Wire-iOS Tests/Notification Service Extension/NotificationServiceTests.swift]: Done

File: Wire-iOS Tests/Notification Service Extension/NotificationServiceTests.swift (hunk 3 above)
Comment (KaterinaWire) [reply Wire-iOS Tests/Notification Service Extension/NotificationServiceTests.swift]: Done

File: Wire-iOS Tests/Notification Service Extension/NotificationServiceTests.swift [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+        return ZMConversation.createTeamGroupConversation(moc: coreDataFixture.uiMOC, otherUser: otherUser, selfUser: selfUser)
+    }
+
+    private func contentHandlerMock(_ content: UNNotificationContent) {
+        contentResult = content
+    }
+
+}
+
+class CallEventProviderMock: CallEventHandlerProtocol {
```
[the remark this answers is not in the document]
Comment (KaterinaWire) [reply Wire-iOS Tests/Notification Service Extension/NotificationServiceTests.swift]: Done

File: Wire Notification Service Extension/NotificationService.swift [hunk 6]
```diff
@@ -180,3 +183,19 @@ extension UNNotificationContent {
     }
 
 }
+
+extension NotificationService: CallEventHandlerInterface {
+
+    func reportIncomingVoIPCall(_ payload: [String: Any]) {
+        if #available(iOS 14.5, *) {
```
[the remark this answers is not in the document]
Comment (KaterinaWire) [reply Wire Notification Service Extension/NotificationService.swift]: I got this error message:
`'reportNewIncomingVoIPPushPayload(_:completion:)' is only available in iOS 14.5 or newer`

**Review by agisilaos — APPROVED** [review]

**Pull request merged**